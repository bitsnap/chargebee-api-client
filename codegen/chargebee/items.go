package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListItems generates chargebee client code to fetch all items
//
// API: https://{site}.chargebee.com/api/v2/items
//
// Documentation: https://apidocs.chargebee.com/docs/api/items?lang=curl#list_items
func GenerateListItems() string {
	return client.GenerateList("Items", "Item", "chargebee.com/api/v2/items")
}

// GenerateRetrieveItem generates chargebee client code to retrieve specific item
//
// API:  https://{site}.chargebee.com/api/v2/items/{item-id}
//
// Documentation: hhttps://apidocs.chargebee.com/docs/api/items?lang=curl#retrieve_an_item
func GenerateRetrieveItem() string {
	return client.GenerateRetrieve("Item", "chargebee.com/api/v2/items")
}

// GenerateItemsModel generates chargebee Item domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/items?lang=curl#item_attributes
func GenerateItemsModel() string {
	return client.GenerateModel("Item", "Item attributes", "https://apidocs.chargebee.com/docs/api/items")
}
