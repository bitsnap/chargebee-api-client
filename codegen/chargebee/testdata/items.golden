# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee items code generation when generating Chargebee API client should be able to generate items models" = '''
(string) (len=1555) "type Item struct {\n    Id string `json:\"id\" validate:\"required\"`\n    Name string `json:\"name\" validate:\"required\"`\n    ExternalName string `json:\"external_name\"`\n    Description string `json:\"description\"`\n    Status enums.StatusEnum `json:\"status\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    ItemFamilyId string `json:\"item_family_id\"`\n    Type enums.TypeEnum `json:\"type\" validate:\"required\"`\n    IsShippable bool `json:\"is_shippable\"`\n    IsGiftable bool `json:\"is_giftable\" validate:\"required\"`\n    RedirectUrl string `json:\"redirect_url\"`\n    EnabledForCheckout bool `json:\"enabled_for_checkout\" validate:\"required\"`\n    EnabledInPortal bool `json:\"enabled_in_portal\" validate:\"required\"`\n    IncludedInMrr bool `json:\"included_in_mrr\"`\n    ItemApplicability enums.ItemApplicabilityEnum `json:\"item_applicability\" validate:\"required\"`\n    GiftClaimRedirectUrl string `json:\"gift_claim_redirect_url\"`\n    Unit string `json:\"unit\"`\n    Metered bool `json:\"metered\" validate:\"required\"`\n    UsageCalculation enums.UsageCalculationEnum `json:\"usage_calculation\"`\n    ArchivedAt *time.Time `json:\"archived_at\"`\n    Channel enums.ChannelEnum `json:\"channel\"`\n    ApplicableItems []ApplicableItem `json:\"applicable_items\"`\n    Id string `json:\"id\"`\n    BundleItems []BundleItem `json:\"bundle_items\"`\n    ItemId string `json:\"item_id\" validate:\"required\"`\n    ItemType enums.ItemTypeEnum `json:\"item_type\"`\n    Quantity int `json:\"quantity\"`\n    PriceAllocation *big.Float `json:\"price_allocation\"`\n}"
'''
